### PR TITLE
168781651 -Users should be able to book an accommodation facility

### PR DESCRIPTION
**Pull request by sabin18**
#### Description of Task to be completed?
- PR to allow a user to book an accommodation
#### How should this be manually tested?
- add booking actions
- add booking reducer
- add booking functionalities
#### Any background context you want to provide?
`N/A`
#### What are the relevant pivotal tracker stories?
[#168781651 ](https://www.pivotaltracker.com/story/show/168781651)
#### Screenshots (if appropriate)
Screen Shot 2019-12-20 at 11 43 29
Screen Shot 2019-12-20 at 11 45 33
#### Questions:
`N/A`

### CODE REVIEW

File: src/components/pages/SingleAccommodation.js [hunk 1]
```diff
@@ -25,34 +35,89 @@ export class SingleAccommodation extends React.Component {
...
+  };
+
+  handleCheckIn = async (checkInDate) => {
+    const checkInValue = dateFnsFormat(checkInDate, 'yyyy-MM-dd');
+    await this.setState((state) => ({ ...state, checkInDate: checkInValue }));
+  }
+
+  handleCheckOut = async (checkOutDate) => {
+    const checkOutValue = dateFnsFormat(checkOutDate, 'yyyy-MM-dd');
+    await this.setState((state) => ({ ...state, checkOutDate: checkOutValue }));
```
Comment (nignanthomas) [inline src/components/pages/SingleAccommodation.js]: If feasible, kindly refactor this functions.

File: src/components/pages/AllAccommodation.js [hunk 2]
```diff
@@ -102,7 +102,7 @@ export class AllAccommodation extends React.Component {
       { checkSupplier() ? (
       <Button href="/accommodations/new">
             <Add />
-            reate new accommodation
+            create new accommodation
```
Comment (nignanthomas) [inline src/components/pages/AllAccommodation.js]: Capital `C`, please.

File: src/components/pages/SingleAccommodation.js [hunk 3]
```diff
@@ -25,34 +35,89 @@ export class SingleAccommodation extends React.Component {
...
+      default:
+        images = imageArray;
+        break;
+    }
+
+    return (
+        <div key={post.id} className="container-fluid single-container ">
           <Row>
         <Col md={5} className="breadcrumbs">
           <Breadcrumbs itemsArray={['> Home', '  accommodations', post.name]} />
```
Comment (nignanthomas) [inline src/components/pages/SingleAccommodation.js]: Capital `A`...

**Review by nignanthomas — CHANGES_REQUESTED** [review]
@sabin18 
Good job here bro

However, kindly note I have left comments and request for changes...
Also we have a `Make Booking` button in `All Accommodations` that does nothing. Kindly find it a functionnality. For example you can attach an hide/show modal that displays the booking component.

Thanks

File: src/styles/pages/singleAccommodation.scss [hunk 4]
```diff
@@ -194,4 +204,7 @@ img{
   width:100%;
   height:370px;
 }
+}
+.disabled-date {
+  pointer-events: none;
 }
```
Comment (ngireric123) [inline src/styles/pages/singleAccommodation.scss:214]: add new line

File: src/tests/singleAccommodationPageTest.spec.js [hunk 5]
```diff
@@ -40,12 +40,57 @@ const setUp = (initialState = {}) => {
...
+    component.find('[data-test="checkout Input"]').simulate('change', checkOutDate);
+    component.find('[data-test="Make Booking"]').simulate('click');
+    component.find('form').simulate('submit', {
+      preventDefault() {},
+      setTimeout() {}
+    });
+    expect(component.find('h3')).toHaveLength(4);
+  });
+  
+
```
Comment (ngireric123) [inline src/tests/singleAccommodationPageTest.spec.js]: remove this line

File: src/components/pages/SingleAccommodation.js [hunk 6]
```diff
@@ -33,35 +42,85 @@ export class SingleAccommodation extends React.Component {
...
+  this.setState({
+    isLoading: false, checkInDate: '', checkOutDate: '', roomsNumber: '',
+  });
+}
+
+renderAcommodation() {
+  const {
+    accommodation, booked, bookedError, bookings,
+  } = this.props;
+  console.log('props', bookedError);
```
Comment (nakiwuge) [inline src/components/pages/SingleAccommodation.js]: Remove console.log

File: src/components/pages/SingleAccommodation.js [hunk 7]
```diff
@@ -33,35 +42,85 @@ export class SingleAccommodation extends React.Component {
...
+  const { userId } = this.state;
+  const post = accommodation;
+  const { ownerUser } = post;
```
Comment (nakiwuge) [inline src/components/pages/SingleAccommodation.js]: Equate this constant to accommodation directly to and remove line 90

File: src/components/pages/SingleAccommodation.js [hunk 8]
```diff
@@ -33,35 +42,85 @@ export class SingleAccommodation extends React.Component {
...
+  const { props, state } = this;
+  const accomodationId = props.accommodation.id;
+  this.setState({ isLoading: true });
+  await props.BookAccommodation(state.checkInDate, state.checkOutDate, accomodationId, state.roomsNumber);
+  this.setState({
+    isLoading: false, checkInDate: '', checkOutDate: '', roomsNumber: '',
+  });
+}
+
+renderAcommodation() {
```
Comment (nakiwuge) [inline src/components/pages/SingleAccommodation.js]: This method is too long kindly create a separate component for bookings and call it in this component

File: src/styles/styles.scss [hunk 9]
```diff
@@ -33,6 +33,9 @@ body {
 input[disabled] {
 	cursor: default;
 }
+body {
+	background-color: rgb(233, 235, 238);
```
Comment (nakiwuge) [inline src/styles/styles.scss]: Put this color in variable s

File: src/tests/actionsTests/bookingActions.spec.js [hunk 10]
```diff
@@ -22,6 +22,13 @@ const bookings = [
     }
 
 ];
+const BookedData = {
+    checkInDate: '2019-12-21',
+	checkOutDate: '2020-01-01',
+	accomodationId: 6,
+	roomsNumber: 1
+
```
Comment (nakiwuge) [inline src/tests/actionsTests/bookingActions.spec.js:30]: Remove this line

**Review by nakiwuge — COMMENTED** [review]
@sabin18 good job sofar. Kindly work on this feedback
- The `ViewSIngleAccommodation` component is too connected each you are supposed to create different components for rating and booking and then call them in this file. Since you are working on booking s create a separate component for bookings.
- When I try to book with invalid dates I get an error which is ok but when I provide valid data the error component still persists so both the previous error and success message are displayed.
- The `Number of rooms` place holder is cut off  in mobile view

**Review by nakiwuge — CHANGES_REQUESTED** [review]
@sabin18 good job sofar. Kindly work on this feedback
- The `ViewSIngleAccommodation` component is too connected each you are supposed to create different components for rating and booking and then call them in this file. Since you are working on booking s create a separate component for bookings.
- When I try to book with invalid dates I get an error which is ok but when I provide valid data the error component still persists so both the previous error and success message are displayed.
- The `Number of rooms` place holder is cut off  in mobile view

**Review by ngireric123 — APPROVED** [review]
LGTM

**Review by nignanthomas — APPROVED** [review]

**Review by nakiwuge — APPROVED** [review]

**Pull request merged**